[MM-46]: Updated the message for exporting channels

Kshitij-Katiyar opened the pull request:
#### Summary
Updated the messages received while exporting different types of channels

### Screenshots
#### Existing
![Screenshot from 2024-08-14 09-19-55](https://github.com/user-attachments/assets/8a6485e0-7102-49d3-8667-cb6abc26524e)
![Screenshot from 2024-08-14 09-05-19](https://github.com/user-attachments/assets/d43555a6-fd68-48ce-b0d2-7aae2e42209e)
![Screenshot from 2024-08-21 14-16-56](https://github.com/user-attachments/assets/be9e99a4-730e-47ec-a16f-9de655ff2d7f)
![Screenshot from 2024-08-21 14-17-40](https://github.com/user-attachments/assets/2aab5978-dffe-421b-8429-8f7f8bf6365c)


#### Updated
![Screenshot from 2024-08-14 09-06-30](https://github.com/user-attachments/assets/97ce7845-f390-454a-ae17-17108755e297)
![Screenshot from 2024-08-21 14-16-20](https://github.com/user-attachments/assets/0ceae8aa-a12f-4ca3-87ab-2b3b39bbb6a8)
![Screenshot from 2024-08-21 16-30-46](https://github.com/user-attachments/assets/f8aacb1e-f238-41cb-a07b-cdf58cd621b9)
![Screenshot from 2024-08-14 09-05-51](https://github.com/user-attachments/assets/661d92e0-c7cc-42fa-9c0d-19b36dd93c27)

raghavaggarwal2308 commented:
@Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?

Kshitij-Katiyar commented:
> @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?

@raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?

raghavaggarwal2308 commented:
> > @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?
> 
> @raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?

Yes, I think we should. Just for consistency

Kshitij-Katiyar commented:
> > > @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?
> > 
> > 
> > @raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?
> 
> Yes, I think we should. Just for consistency

@raghavaggarwal2308 We have implemented your suggestions and updated the screenshot accordingly. Please have a look

raghavaggarwal2308 commented on server/command_hooks.go:
```diff
@@ -88,6 +88,35 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
+		channelToExportName = channelToExport.Name
+		team, err := p.API.GetTeam(args.TeamId)
+		if err != nil {
+			p.client.Log.Error("error occurred while getting the details of team for the channel.",
+				"GM channel ID", args.ChannelId, "User ID", args.UserId, "Error", err)
```
```suggestion
			p.client.Log.Error("error occurred while getting the details of team for the channel.", "GMChannelID", args.ChannelId, "TeamID", args.TeamId, "Error", err)
```

raghavaggarwal2308 commented on server/command_hooks.go:
```diff
@@ -88,6 +88,35 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
+			p.client.Log.Error("error occurred while getting the details of team for the channel.",
+				"GM channel ID", args.ChannelId, "User ID", args.UserId, "Error", err)
+
+			return &model.CommandResponse{
+				ResponseType: model.CommandResponseTypeEphemeral,
+				Text:         fmt.Sprintf("An error occurred trying to exporting the chat for %s GM channel", channelToExportName),
+			}
```
Having this link is not a must so we can just log the error and continue with our functionality without a link

raghavaggarwal2308 commented on server/command_hooks.go:
```diff
@@ -88,6 +88,48 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
+		exportProcessMsg = fmt.Sprintf("Exporting ~%s. @%s will send you a direct message when the export is ready.", channelToExportName, botUsername)
+	case model.ChannelTypeGroup:
+		channelToExportName = channelToExport.Name
+		team, appErr := p.API.GetTeam(args.TeamId)
```
use p.client

raghavaggarwal2308 replied:
discussed

Kshitij-Katiyar commented:
Created a PR with all the changes on mattermost repo, hence closing this one.
